pkg/sysfs/system.go

```diff
@@ -1526,6 +1554,88 @@ func (sys *system) discoverSst() error {
 	return nil
 }
 
+func parseSize(s string) (uint64, error) {
+	var size uint64
+	var unit string
+	if _, err := fmt.Sscanf(s, "%d%s", &size, &unit); err != nil {
+		if _, err := fmt.Sscanf(s, "%d", &size); err != nil {
+			return 0, err
+		}
+	}
+	switch unit {
+	case "k", "K", "KB":
+		size *= 1 << 10
+	case "M", "MB":
+		size *= 1 << 20
+	case "G", "GB":
+		size *= 1 << 30
+	case "T", "TB":
+		size *= 1 << 40
+	case "":
+	default:
+		return 0, fmt.Errorf("unknown unit %q", unit)
+	}
+	return size, nil
+}
+
+// Parse cache overrides from a JSON string.
+func parseCpuCacheOverrides(overrideJson string) map[int][]*Cache {
```

Comment:
nit: consider changing this to return signature of `(map[int][]*Cache, error)` and return a `nil, fmt.Errorf("failed to parse CPU cache overrides: %w", err)` in case of a parsing or some other error, to be consistent with how core kind overrides work (IIRC).

Reply:
Return value changed.

Moved the parse function to be called only on demand as part of discoverCPU() and next to discoverCache(). This enabled making it sys.parseCpuCacheOverrides(), so I can use the same Debug() as everywhere else.

I think it's much better now. Thank you for these comments, once again!